### ISSUE TITLE
aur/ package name prefix combined with -a flag causes a sed error

### ISSUE THREAD

**Issue by vith** [rmarquis/pacaur#740]
###### Version

```
pacaur 4.7.10
```

###### Description

`aur/` package name prefix combined with `-a` flag breaks the command. Removing either or both avoids the error.

###### Output

`pacaur -Sa aur/gnome-shell-extension-no-topleft-hot-corner`

```
:: resolving dependencies...
sed: -e expression #1, char 15: extra characters after command
sed: -e expression #1, char 15: extra characters after command
sed: -e expression #1, char 15: extra characters after command
sed: -e expression #1, char 15: extra characters after command
:: no results found for aur/gnome-shell-extension-no-topleft-hot-corner
```

###### Debug output

`bash -x pacaur -Sa aur/gnome-shell-extension-no-topleft-hot-corner`

```
+ version=4.7.10
+ unset aur cleancache devel edit info installpkg foreign needed noconfirm nodeps noedit
+ unset operation pac pacQ pacS quiet rebuild refresh repo search selective upgrade
+ LC_COLLATE=C
+ TEXTDOMAIN=pacaur
+ TEXTDOMAINDIR=/usr/share/locale
+ [[ -n '' ]]
+ configdir=/etc/xdg/pacaur
+ userconfigdir=/home/vith/.config/pacaur
+ userpacmandir=/home/vith/.config/pacman
+ usercachedir=/home/vith/.cache/pacaur
+ tmpdir=/run/user/1000
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -r '' ]]
+ source /etc/makepkg.conf
++ DLAGENTS=('ftp::/usr/bin/curl -fC - --ftp-pasv --retry 3 --retry-delay 3 -o %o %u' 'http::/usr/bin/curl -fLC - --retry 3 --retry-delay 3 -o %o %u' 'https::/usr/bin/curl -fLC - --retry 3 --retry-delay 3 -o %o %u' 'rsync::/usr/bin/rsync --no-motd -z %u %o' 'scp::/usr/bin/scp -C %u %o')
++ VCSCLIENTS=('bzr::bzr' 'git::git' 'hg::mercurial' 'svn::subversion')
++ CARCH=x86_64
++ CHOST=x86_64-pc-linux-gnu
++ CPPFLAGS=-D_FORTIFY_SOURCE=2
++ CFLAGS='-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt'
++ CXXFLAGS='-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt'
++ LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
+++ nproc
++ MAKEFLAGS=-j4
++ DEBUG_CFLAGS='-g -fvar-tracking-assignments'
++ DEBUG_CXXFLAGS='-g -fvar-tracking-assignments'
++ BUILDENV=(!distcc color !ccache check !sign)
++ OPTIONS=(strip docs !libtool !staticlibs emptydirs zipman purge !optipng !upx !debug)
++ INTEGRITY_CHECK=(md5)
++ STRIP_BINARIES=--strip-all
++ STRIP_SHARED=--strip-unneeded
++ STRIP_STATIC=--strip-debug
++ MAN_DIRS=({usr{,/local}{,/share},opt/*}/{man,info})
++ DOC_DIRS=(usr/{,local/}{,share/}{doc,gtk-doc} opt/*/{doc,gtk-doc})
++ PURGE_TARGETS=(usr/{,share}/info/dir .packlist *.pod)
++ COMPRESSGZ=(gzip -c -f -n)
++ COMPRESSBZ2=(bzip2 -c -f)
++ COMPRESSXZ=(xz -c -z -)
++ COMPRESSLRZ=(lrzip -q)
++ COMPRESSLZO=(lzop -q)
++ COMPRESSZ=(compress -c -f)
++ PKGEXT=.pkg.tar.xz
++ SRCEXT=.src.tar.gz
+ [[ -r /home/vith/.config/pacman/makepkg.conf ]]
+ [[ -r /home/vith/.makepkg.conf ]]
+ PACMAN=
+ PKGDEST=
+ SRCDEST=
+ SRCPKGDEST=
+ LOGDEST=
+ BUILDDIR=
+ PKGEXT=.pkg.tar.xz
+ SRCEXT=.src.tar.gz
+ GPGKEY=
+ PACKAGER=
+ CARCH=x86_64
+ editor=vi
+ displaybuildfiles=diff
+ fallback=true
+ silent=false
+ sortby=popularity
+ sortorder=descending
+ sudoloop=true
+ pacmanbin=pacman
+ clonedir=/home/vith/.cache/pacaur
+ aururl=aur.archlinux.org
+ aurrpc='/rpc/?type=info&v=5'
+ source /etc/xdg/pacaur/config
++ displaybuildfiles=none
+ [[ -r /home/vith/.config/pacaur/config ]]
+ [[ ! -d /home/vith/.cache/pacaur ]]
+ declare -A jsoncache
+ trap Cancel INT
+ args=($@)
+ for i in "${args[@]}"
+ [[ -Sa =~ ^-[a-zA-Z0-9] ]]
+ opts+=($i)
+ for i in "${args[@]}"
+ [[ aur/gnome-shell-extension-no-topleft-hot-corner =~ ^-[a-zA-Z0-9] ]]
+ count=0
+ [[ -n -Sa ]]
+ getopts sidmykufecqrahvxVDFQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ pacS=1
+ operation=sync
+ [[ -Sa =~ w ]]
+ installpkg=true
+ [[ -Sa =~ g ]]
+ [[ -Sa =~ l ]]
+ [[ -Sa =~ p ]]
+ getopts sidmykufecqrahvxVDFQRSTUbglnoptw-: OPT
+ pacmanarg+=("-$OPT")
+ case "$OPT" in
+ aur=1
+ getopts sidmykufecqrahvxVDFQRSTUbglnoptw-: OPT
+ [[ -z aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ pkgs+=("${!OPTIND}")
+ shift 2
+ OPTIND=1
+ [[ -n '' ]]
++ grep '^Color' /etc/pacman.conf
+ [[ -n Color ]]
+ [[ '' != \n\e\v\e\r ]]
+ [[ '' = \a\l\w\a\y\s ]]
+ coweropts+=("--color=auto")
+ reset='\e[0m'
+ colorR='\e[1;31m'
+ colorG='\e[1;32m'
+ colorY='\e[1;33m'
+ colorB='\e[1;34m'
+ colorM='\e[1;35m'
+ colorC='\e[1;36m'
+ colorW='\e[1;39m'
+ pacmanarg=(${pacmanarg[@]/--/})
+ pacmanarg=(${pacmanarg[@]/-r/})
+ pacmanarg=(${pacmanarg[@]/-a/})
+ [[ sync = sync ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ [[ 1000 -eq 0 ]]
+ [[ -n 1 ]]
+ pacmanarg=(${pacmanarg[@]/-e/})
+ [[ -n 1 ]]
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ [[ -n '' ]]
++ command -v vi
+ [[ ! -n /usr/bin/vi ]]
++ basename pacaur
+ [[ '' = pacaur ]]
+ [[ ! -w /home/vith/.cache/pacaur ]]
+ [[ -z aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ [[ -z aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ [[ -n '' ]]
+ case $operation in
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -z aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ ClassifyPkgs aur/gnome-shell-extension-no-topleft-hot-corner
+ local noaurpkgs norepopkgs
+ [[ true = true ]]
+ [[ -n '' ]]
+ [[ -n 1 ]]
+ aurpkgs=(${pkgs[@]})
+ [[ ! -n '' ]]
+ [[ ! -n 1 ]]
+ [[ -n '' ]]
+ [[ -n aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ [[ -n '' ]]
+ [[ true = true ]]
+ [[ ! -n 1 ]]
+ Core
+ GetIgnoredPkgs
+ ignoredpkgs+=($(grep '^IgnorePkg' '/etc/pacman.conf' | awk -F '=' '{print $NF}' | tr -d "'\""))
++ grep '^IgnorePkg' /etc/pacman.conf
++ awk -F = '{print $NF}'
++ tr -d ''\''"'
+ [[ -e /home/vith/.config/cower/config ]]
+ ignoredpkgs=(${ignoredpkgs[@]//,/ })
+ GetIgnoredGrps
+ ignoredgrps+=($(grep '^IgnoreGroup' '/etc/pacman.conf' | awk -F '=' '{print $NF}' | tr -d "'\""))
++ grep '^IgnoreGroup' /etc/pacman.conf
++ awk -F = '{print $NF}'
++ tr -d ''\''"'
+ ignoredgrps=(${ignoredgrps[@]//,/ })
+ [[ -n '' ]]
+ IgnoreChecks
+ local checkaurpkgs checkaurpkgsAver checkaurpkgsAgrp checkaurpkgsQver checkaurpkgsQgrp i json
+ [[ -z '' ]]
+ [[ -z '' ]]
+ return
+ DepsSolver
+ local i aurpkgsname aurpkgsver aurpkgsaurver aurpkgsconflicts
+ Note i 'resolving dependencies...'
+ case "$1" in
+ echo -e '\e[1;34m::\e[0m resolving dependencies...'
:: resolving dependencies...
+ for i in "${!aurpkgs[@]}"
++ awk -F '>|<|=' '{print $1}'
+ aurpkgsnover[$i]=aur/gnome-shell-extension-no-topleft-hot-corner
+ SetJson aur/gnome-shell-extension-no-topleft-hot-corner
+ [[ 1 -eq 0 ]]
+ [[ -z '' ]]
++ DownloadJson aur/gnome-shell-extension-no-topleft-hot-corner
++ local urlencodedpkgs urlargs urlcurl urlarg urlmax j
++ urlencodedpkgs=($(sed 's/+/%2b/g;s/@/%40/g' <<< $@))
+++ sed 's/+/%2b/g;s/@/%40/g'
++ urlarg='&arg[]='
+++ printf '&arg[]=%s' aur/gnome-shell-extension-no-topleft-hot-corner
++ urlargs='&arg[]=aur/gnome-shell-extension-no-topleft-hot-corner'
++ urlmax=4400
++ [[ 54 -lt 4400 ]]
++ curl -sfg --compressed -C 0 'https://aur.archlinux.org/rpc/?type=info&v=5&arg[]=aur/gnome-shell-extension-no-topleft-hot-corner'
+ jsoncache[$@]='{"version":5,"type":"multiinfo","resultcount":0,"results":[]}'
+ json='{"version":5,"type":"multiinfo","resultcount":0,"results":[]}'
+ aurpkgsproviders=(${aurpkgsnover[@]})
+ aurpkgsproviders+=($(GetJson "array" "$json" "Provides"))
++ GetJson array '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Provides
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/^"Provides"/,/]/!d'
++ cut '-d ' -f 2-
++ sed 's/] /]\n/g'
++ tr '\n' ' '
++ tr -d '\n'
++ tr -d '[]"'
++ tr -d '\"'
+ for i in "${!aurpkgsproviders[@]}"
++ awk -F '>|<|=' '{print $1}'
+ aurpkgsproviders[$i]=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurpkgsconflicts=($(GetJson "array" "$json" "Conflicts"))
++ GetJson array '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Conflicts
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/^"Conflicts"/,/]/!d'
++ tr '\n' ' '
++ sed 's/] /]\n/g'
++ tr -d '\"'
++ cut '-d ' -f 2-
++ tr -d '[]"'
++ tr -d '\n'
+ [[ -n '' ]]
+ deps=(${aurpkgsnover[@]})
+ [[ -z '' ]]
+ foreignpkgs=($($pacmanbin -Qmq))
++ pacman -Qmq
+ FindDepsAur aur/gnome-shell-extension-no-topleft-hot-corner
+ local depspkgs depspkgstmp depspkgsaurtmp repodepstmp builtpkg vcsdepspkgs assumedepspkgs
+ local aurversionpkgs aurversionpkgsname aurversionpkgsver aurversionpkgsaurver i j json
+ [[ -n '' ]]
+ unset aurversionpkgs
+ [[ -z '' ]]
+ SetJson aur/gnome-shell-extension-no-topleft-hot-corner
+ [[ 1 -eq 0 ]]
+ [[ -z {"version":5,"type":"multiinfo","resultcount":0,"results":[]} ]]
+ json='{"version":5,"type":"multiinfo","resultcount":0,"results":[]}'
+ aurversionpkgs=(${aurpkgs[@]})
+ [[ -n aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ for i in "${!aurversionpkgs[@]}"
+ unset aurversionpkgsname aurversionpkgsver aurversionpkgsaurver
+ aurversionpkgsname=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurversionpkgsname=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurversionpkgsname=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurversionpkgsver=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurversionpkgsver=aur/gnome-shell-extension-no-topleft-hot-corner
+ aurversionpkgsver=aur/gnome-shell-extension-no-topleft-hot-corner
++ GetJson varvar '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Version aur/gnome-shell-extension-no-topleft-hot-corner
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/"Name":"aur/gnome-shell-extension-no-topleft-hot-corner"/,/}/!d'
++ grep -Po 'Version:.*'
++ sed -r s/Version:/Version#/g
++ awk -F '#' '{print $2}'
sed: -e expression #1, char 15: extra characters after command
++ tr -d '"'
+ aurversionpkgsaurver=
+ [[ ! -n '' ]]
+ [[ !    =~  aur/gnome-shell-extension-no-topleft-hot-corner  ]]
+ errdeps+=(${aurversionpkgs[$i]})
+ continue
+ depspkgs=($(GetJson "array" "$json" "Depends"))
++ GetJson array '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Depends
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/^"Depends"/,/]/!d'
++ tr '\n' ' '
++ sed 's/] /]\n/g'
++ tr -d '\n'
++ tr -d '\"'
++ cut '-d ' -f 2-
++ tr -d '[]"'
+ [[ ! -n '' ]]
+ depspkgs+=($(GetJson "array" "$json" "MakeDepends"))
++ GetJson array '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' MakeDepends
++ json_verify -q
++ case "$1" in
++ json_reformat
++ sed -e '/^"MakeDepends"/,/]/!d'
++ tr -d ', '
++ tr '\n' ' '
++ sed 's/] /]\n/g'
++ tr -d '\"'
++ cut '-d ' -f 2-
++ tr -d '[]"'
++ tr -d '\n'
+ depspkgs+=($(GetJson "array" "$json" "CheckDepends"))
++ GetJson array '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' CheckDepends
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/^"CheckDepends"/,/]/!d'
++ tr -d '\n'
++ tr '\n' ' '
++ sed 's/] /]\n/g'
++ tr -d '\"'
++ cut '-d ' -f 2-
++ tr -d '[]"'
+ [[ -n aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ depspkgs=($(grep -xvf <(printf '%s\n' "${aurpkgsproviders[@]}") <(printf '%s\n' "${depspkgs[@]}")))
++ grep -xvf /dev/fd/63 /dev/fd/62
+++ printf '%s\n' aur/gnome-shell-extension-no-topleft-hot-corner
+++ printf '%s\n'
+ [[ x86_64 == \i\6\8\6 ]]
+ [[ ! -n '' ]]
+ [[ ! -n '' ]]
+ depspkgs=($($pacmanbin -T ${depspkgs[@]} | sort -u))
++ pacman -T
++ sort -u
+ unset depspkgsaur
+ [[ -n '' ]]
+ unset depspkgs
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ deps=($(grep -xvf <(printf '%s\n' "${aurdepspkgs[@]}") <(printf '%s\n' "${deps[@]}")))
++ grep -xvf /dev/fd/63 /dev/fd/62
+++ printf '%s\n'
+++ printf '%s\n' aur/gnome-shell-extension-no-topleft-hot-corner
+ deps+=(${aurdepspkgs[@]})
+ SetJson aur/gnome-shell-extension-no-topleft-hot-corner
+ [[ 1 -eq 0 ]]
+ [[ -z {"version":5,"type":"multiinfo","resultcount":0,"results":[]} ]]
+ json='{"version":5,"type":"multiinfo","resultcount":0,"results":[]}'
+ SortDepsAur aur/gnome-shell-extension-no-topleft-hot-corner
+ local i j sortaurpkgs sortdepspkgs sortdepspkgsaur
+ [[ -z '' ]]
+ sortaurpkgs=(${aurpkgs[@]})
+ unset checkedsortdepspkgsaur
+ for i in "${!sortaurpkgs[@]}"
+ unset sortdepspkgs sortdepspkgsaur
+ sortdepspkgs+=($(GetJson "arrayvar" "$json" "Depends" "${sortaurpkgs[$i]}"))
++ GetJson arrayvar '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Depends aur/gnome-shell-extension-no-topleft-hot-corner
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ tr -d '\"'
++ sed -e '/"Name":"aur/gnome-shell-extension-no-topleft-hot-corner"/,/}/!d'
++ tr '\n' ' '
++ sed -e '/^"Depends"/,/]/!d'
++ cut '-d ' -f 2-
++ tr -d '[]'
sed: -e expression #1, char 15: extra characters after command
+ sortdepspkgs+=($(GetJson "arrayvar" "$json" "MakeDepends" "${sortaurpkgs[$i]}"))
++ GetJson arrayvar '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' MakeDepends aur/gnome-shell-extension-no-topleft-hot-corner
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d ', '
++ sed -e '/"Name":"aur/gnome-shell-extension-no-topleft-hot-corner"/,/}/!d'
++ tr '\n' ' '
++ tr -d '\"'
++ cut '-d ' -f 2-
++ tr -d '[]'
sed: -e expression #1, char 15: extra characters after command
++ sed -e '/^"MakeDepends"/,/]/!d'
+ sortdepspkgs+=($(GetJson "arrayvar" "$json" "CheckDepends" "${sortaurpkgs[$i]}"))
++ GetJson arrayvar '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' CheckDepends aur/gnome-shell-extension-no-topleft-hot-corner
++ json_verify -q
++ case "$1" in
++ json_reformat
++ sed -e '/"Name":"aur/gnome-shell-extension-no-topleft-hot-corner"/,/}/!d'
++ tr -d '\"'
++ tr '\n' ' '
++ tr -d ', '
++ sed -e '/^"CheckDepends"/,/]/!d'
sed: -e expression #1, char 15: extra characters after command
++ cut '-d ' -f 2-
++ tr -d '[]'
+ for j in "${!errdeps[@]}"
++ awk -F '>|<|=' '{print $1}'
+ errdepsnover[$j]=aur/gnome-shell-extension-no-topleft-hot-corner
+ [[ -z '' ]]
+ tsortdeps+=("${sortaurpkgs[$i]} ${sortaurpkgs[$i]}")
+ sortdepspkgsaur=($(grep -xvf <(printf '%s\n' "${allcheckedsortdepspkgsaur[@]}") <(printf '%s\n' "${sortdepspkgsaur[@]}")))
++ grep -xvf /dev/fd/63 /dev/fd/62
+++ printf '%s\n'
+++ printf '%s\n'
+ [[ -n '' ]]
+ [[ -n '' ]]
+ deps=($(tsort <<< ${tsortdeps[@]}))
++ tsort
+ (( 0 > 0 ))
+ depsAname=($(GetJson "var" "$json" "Name"))
++ GetJson var '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Name
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d '", '
++ grep -Po 'Name:.*'
++ awk -F '#' '{print $2}'
++ sed -r s/Name:/Name#/g
+ depsAver=($(GetJson "var" "$json" "Version"))
++ GetJson var '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Version
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d '", '
++ grep -Po 'Version:.*'
++ awk -F '#' '{print $2}'
++ sed -r s/Version:/Version#/g
+ depsAood=($(GetJson "var" "$json" "OutOfDate"))
++ GetJson var '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' OutOfDate
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d '", '
++ grep -Po 'OutOfDate:.*'
++ sed -r s/OutOfDate:/OutOfDate#/g
++ awk -F '#' '{print $2}'
+ depsAmain=($(GetJson "var" "$json" "Maintainer"))
++ GetJson var '{"version":5,"type":"multiinfo","resultcount":0,"results":[]}' Maintainer
++ json_verify -q
++ case "$1" in
++ json_reformat
++ tr -d '", '
++ grep -Po 'Maintainer:.*'
++ sed -r s/Maintainer:/Maintainer#/g
++ awk -F '#' '{print $2}'
+ [[ -n aur/gnome-shell-extension-no-topleft-hot-corner ]]
+ for i in "${!errdepsnover[@]}"
+ [[  aur/gnome-shell-extension-no-topleft-hot-corner  =~  aur/gnome-shell-extension-no-topleft-hot-corner  ]]
+ Note f 'no results found for aur/gnome-shell-extension-no-topleft-hot-corner'
+ case "$1" in
+ echo -e '\e[1;31m::\e[0m no results found for aur/gnome-shell-extension-no-topleft-hot-corner'
:: no results found for aur/gnome-shell-extension-no-topleft-hot-corner
+ exit 1
```

**Comment by rmarquis** [comment]
Indeed. The `ClassifyPkgs()` function should be reworked.
Thanks for the report!

**Comment by rmarquis** [comment]
Fixed in 7b598e5a519b69a38ee1c725564b37ddca0c4307.
